doc/src/build/fullnode.md

```diff
@@ -288,3 +288,97 @@ If you followed the instructions for [Building from Source](#building-from-sourc
+  # older than the current epoch. In general, when this is N (where N >= 1), pruner 
+  # will only prune object versions from checkpoints upto current - N epoch. 
+  # This means it is possible to have multiple versions of an object to be present 
+  # in the db. It is a recommended setting for fullnodes as they may have to serve 
+  # rpc requests which require looking up objects by id and version (instead of just latest
+  # version). However, if a fullnode is not serving such rpc requests it should also 
+  # use aggressive pruning
+  num-epochs-to-retain: 0
+  # Advanced setting: Maximum number of checkpoints in one batch of pruning run. This is
+  # mostly not needed to be changed and default is good for most use cases
+  max-checkpoints-in-batch: 10
+  # Advanced setting: Maximum number of transactions in one batch of pruning run. This is
+  # mostly not needed to be changed and default is good for most use cases
+  max-transactions-in-batch: 1000
+```
+## Transaction Pruning
+
+Transaction pruning is another type of pruning that removes old transactions and effects from the db. 
+Checkpoints which are created periodically contain transaction and their effects. Transaction pruning
+works in the background by tailing checkpoint stream as well. A sui node can enable transaction pruning
+by adding `num_epochs_to_retain_for_checkpoints` to the `authority-store-pruning-config` config:
+
+```yaml
+authority-store-pruning-config:
+  num-latest-epoch-dbs-to-retain: 3
+  epoch-db-pruning-period-secs: 3600
+  num-epochs-to-retain: 0
+  max-checkpoints-in-batch: 10
+  max-transactions-in-batch: 1000
+  # Number of latest epochs before which transaction and effects could be pruned.
+  # When this is N (where N >= 2), pruner can prune transaction and effects from 
+  # checkpoints upto current - N epoch. Transaction and effects from current and
+  # previous epoch are never pruned. N = 2 is a recommended setting for validators
+  # and fullnodes not serving rpc requests.
+  num_epochs_to_retain_for_checkpoints: 2
+  # Ensures that individual SST files periodically go through the compaction process.
+  # This helps reclaim disk space and avoid fragmentation issues
+  periodic-compaction-threshold-days: 1
+```
+
+## Archival Fallback
+
+Once various full nodes start pruning historical transactions and effects, it may not be always possible for
+their peers to catch up transaction and effects via state synchronization. And for this a node needs to fallback
+downloading this data from a remote aws s3 bucket. This archive has all transaction and effects data since 
+genesis and lags behind the latest network checkpoint by ten minutes. Enabling this is very highly recommended
+for all nodes. A sui node can enable this automatic archive fallback by adding the following config:
+
+```yaml
+state-archive-read-config:
+  - object-store-config:
+      object-store: "S3"
+      # Use mysten-testnet-archives for testnet 
```

Comment:
Can we add something on enabling a fullnode to create it's own archive? So ppl don't have to use mysten's buckets if they dont want to

Reply:
sure, will add